dev-docs/metrics.md

```diff
@@ -234,3 +234,115 @@ Strong references to instruments will be discarded when changes to the aggregate
 On the fast path the mutex is locked for the period that it takes to upgrade the weak reference. This is a fast operation, and should not block the thread for any meaningful period of time.
 
 If there is shown to be contention in future profiling we can revisit.
+
+## Adding new metrics
+There are different types of metrics.
+
+* Static - Declared via macro, cannot be configured, low cardinality, and are transmitted to Apollo.
+* Dynamic - Configurable via yaml, not transmitted to Apollo.
+
+> New features should add BOTH static and dynamic metrics.
+
+> Why are static metrics less good for users to for debugging?
+> 
+> They can be used, but usually it'll be only a starting point for them. We can't predict the things that users will want to monitor, and if we tried we would blow up the cardinality of our metrics resulting in high costs for our users via their APMs.
+> 
+> For instance, it we must not add operation name to the attributes of a static metric as this is potentially infinite cardinality, but as a dynamic metric this is fine as users can use conditions to reduce the amount of data they are looking at.
+
+### Naming
+Metrics should be named in a way that is consistent with the rest of the metrics in the system.
+
+**Metrics**
+* `<feature>` - This should be a noun that describes the feature that the metric is monitoring.
+
+* `<feature>.<verb>` - Sub-metrics are usually a verb that describes the action that the metric is monitoring.
+
+**Attributes**
+* `<feature>.<feature-specific-attribute>` - Are always prefixed with the feature name unless they are standard metrics from the [otel semantic conventions](https://opentelemetry.io/docs/specs/semconv/general/metrics/). 
+
+### Static metrics
+When adding a new feature to the Router you must also add new static metrics to monitor the usage of that feature and users cannot turn them off.
+These metrics must be low cardinality and not leak any sensitive information. Users cannot change the attributes that are attached to these metrics.
+These metrics are transmitted to Apollo unless explicitly disabled.
+
+When adding new static metrics and attributes make sure to:
+* Include them in your design document.
+* Look at the [OTel semantic conventions](https://opentelemetry.io/docs/specs/semconv/general/metrics/) 
+* Notify `#proj-router-analytics` channel in Slack.
+* Add the metrics to the spreadsheet linked in the `#proj-router-analytics` channel in Slack.
+
+To define a static metric us a macro:
+```rust
+u64_counter!("apollo.router.<feature>.<verb>", "description", 1, "attr" => "val");
```

Comment:
We now have a new kind of macros to specify unit, it's `u64_counter_with_unit` I think and we should always provide units

Reply:
We should also remove any non unit macros

Reply:
[35575dc](https://github.com/apollographql/router/pull/5116/commits/35575dc7daf9e616d2b5e8d86c757857f0c1f375)